src/Remote/Composer.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Castor\Remote;
+
+use Castor\Console\Application;
+use Castor\GlobalHelper;
+use Castor\PathHelper;
+use Castor\Remote\Exception\ComposerError;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Helper\ProgressIndicator;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\ExecutableFinder;
+use Symfony\Component\Process\Process;
+
+use function Castor\fingerprint;
+
+class Composer
+{
+    public const VENDOR_DIR = '/.castor/vendor/';
+    public const DEFAULT_COMPOSER_CONFIGURATION = [
+        'description' => 'This file is managed by Castor. Do not edit it manually.',
+        'config' => [
+            'sort-packages' => true,
+        ],
+        'replace' => [
+            'castor/castor' => Application::VERSION,
+        ],
+    ];
+
+    public function __construct(
+        private readonly Filesystem $filesystem,
+        private readonly LoggerInterface $logger,
+        /** @var array<string, mixed> */
+        private array $configuration = self::DEFAULT_COMPOSER_CONFIGURATION,
+    ) {
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    public function getConfiguration(): array
+    {
+        return $this->configuration;
+    }
+
+    /**
+     * @param array<string, mixed> $configuration
+     */
+    public function setConfiguration(array $configuration): void
+    {
+        $this->configuration = $configuration;
+    }
+
+    public function update(bool $force = false, bool $displayProgress = true): void
+    {
+        $composer = (new ExecutableFinder())->find('composer');
+
+        if (!$composer) {
+            throw new ComposerError('The "composer" executable was not found. In order to use remote import, please make sure that Composer is installed and available in your PATH.');
+        }
+
+        $dir = PathHelper::getRoot() . self::VENDOR_DIR;
+
+        if (!is_dir($dir)) {
+            mkdir($dir, recursive: true);
+        }
```

Comment:
let's use symfony/fs

```suggestion
        $this->fs->mkdir($dir);
```